### ISSUE TITLE
Font name is parsed incorrectly which produces assertion error in debug mode

### ISSUE THREAD

**Issue by MartinHlavna** [DavBfr/dart_pdf#798]
**Describe the bug**
Hello,
I am using FreeSans.ttf font. In debug mode, pdf generation fails because of assertion at data_types.dart:369.

``` assert(c < 0xff && c > 0x00); ```

Looks like font name is parsed incorrectly because value is "/reeSans" and value.codeUnits ends with 0.

**Expected behavior**
Code should not crash :)

**Flutter Doctor**

```
[✓] Flutter (Channel unknown, 2.2.3, on Linux, locale sk_SK.UTF-8)
    • Flutter version 2.2.3 at /opt/flutter-master
    • Framework revision f4abaa0735 (2 months ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/opt/android-sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /home/opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 4.1)
    • Android Studio at /home/opt/android-studio
    • Flutter plugin version 52.2.1
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Android Studio
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2)
    • IntelliJ at /home/opt/idea
    • Flutter plugin version 58.0.4
    • Dart plugin version 212.4746.57

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3)
    • IntelliJ at /opt/idea
    • Flutter plugin version 55.0.4
    • Dart plugin version 203.7759

[✓] Connected device (2 available)
    • SM G398FN (mobile) • R58M87WX64E • android-arm64 • Android 11 (API 30)
    • Linux (desktop)    • linux       • linux-x64     • Linux

! Doctor found issues in 1 category.
```

**Smartphone (please complete the following information):**
 - Device: [Samsung XCover 3S ]
 - OS: [Android 11]

**Comment by MartinHlavna** [comment]
Hi,
this is the two fonts I am using, one bold and one normal

[fonts.tar.gz](https://github.com/DavBfr/dart_pdf/files/7152266/fonts.tar.gz)

Thanks

**Comment by DavBfr** [comment]
Fixed in pdf-3.6.0
Don't hesitate to [Buy Me A Coffee](https://www.buymeacoffee.com/JORBmbw9h).

**Comment by MartinHlavna** [comment]
Thanks!
